MovieQuiz/Services/MoviesLoader.swift

```diff
@@ -6,33 +6,36 @@ protocol MoviesLoading {
 
 struct MoviesLoader: MoviesLoading {
     // MARK: - NetworkClient
-    private let networkClient = NetworkClient()
+    private let networkClient: NetworkRouting
+    
+    init(networkClient: NetworkRouting = NetworkClient()) {
+        self.networkClient = networkClient
+    }
     
     // MARK: - URL
     private var mostPopularMoviesUrl: URL {
-        
         guard let url = URL(string: "https://imdb-api.com/en/API/Top250Movies/k_h7m0bhlk") else {
-            preconditionFailure("Невозможно преобразовать mostPopularMoviesUrl")
+            preconditionFailure("Unable to construct mostPopularMoviesUrl")
         }
         return url
     }
     
+    private enum DecodingError: Error {
+        case errorInCode
+    }
+    
     func loadMovies(handler: @escaping (Result<MostPopularMovies, Error>) -> Void) {
-        networkClient.fetch(url: mostPopularMoviesUrl){ result in
-            switch result{
-            case.success(let data):
-                do
-                {
-                    let movies = try JSONDecoder().decode(MostPopularMovies.self, from: data)
-                    handler(.success(movies))
-                }catch let error{
-                    handler(.failure(error))
-                }
-            case.failure(let error):
-                handler(.failure(error))
+        networkClient.fetch(url: mostPopularMoviesUrl) { result in
+            switch result {
+            case .failure(let error): handler(.failure(error))
+            case .success(let data):
+                let topMovieList = try? JSONDecoder().decode(MostPopularMovies.self, from: data)
+                if let topMovieList = topMovieList {
+                    handler(.success(topMovieList)) } else {
+                        handler(.failure(DecodingError.errorInCode))
+                    }
```

Comment:
Форматирование стоит соблюдать :)
```suggestion
                if let topMovieList = topMovieList {
                    handler(.success(topMovieList)) 
                } else {
                    handler(.failure(DecodingError.errorInCode))
                }
```